### PR TITLE
Linkedlist

### PR DESCRIPTION
**Pull request by peonicles**
@cbas @zellwk @sevresbabylone

Totally not reinventing the wheel here with my linked list implementation

### CODE REVIEW

File: public/js/LinkedList.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        return -1
+      } else {
+        // data is at head
+        // reset LinkedList properties to null and return
+        previousNode.data = null // destroy data
+        this.head = null
+        this.tail = null
+        this.length = 0
+        // this.print()
+        return data
```
Comment (cbas) [inline public/js/LinkedList.js]: What if `data` is `-1`?

File: public/js/LinkedList.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    // if .next === null, we are at tail, the last node
+    // print tail node data, then exit
+    nodeDataArr.push(currentNode.data)
+    // console.log('converting to array:', nodeDataArr)
+
+    return nodeDataArr
+  }
+}
+
+// Tests
```
Comment (cbas) [inline public/js/LinkedList.js]: Add some assertions to make them proper test cases?

**Comment by cbas** [comment]
Was trying to understand the `remove()` method but my brain exploded ... could you add tests instead so I can just look at examples that demonstrate the edge cases? :sweat_smile:

**Comment by peonicles** [comment]
@cbas my current tests are below (commented out). Used the the console.log library ...

I'll take a stab at using mocha / chai in a new PR

File: public/js/LinkedList.js (hunk 1 above)
Comment (peonicles) [inline public/js/LinkedList.js]: I considered returning `NaN` instead to accomodate `data === -1` , not sure if that's good practice though ...

Comment (cbas) [inline public/js/LinkedList.js]: In terms of API design I'd go with:
- Return `true`/`false` depending on whether or not the data was removed.
- Always return `true` since not removing the data is the same as removing it.
- Return the list itself to allow chaining method calls.
- Return `undefined` when nothing was returned, `data` otherwise.
- ...

Comment (peonicles) [inline public/js/LinkedList.js]: Ok, returning the list for chaining sounds like a good idea.
